### DISCUSSION TITLE
Visualising crystal orientations

### DISCUSSION THREAD

**Discussion by FelixTheTBone** [neperfepx/neper#316] [Q&A] [answered]
Hi there, many thanks for providing such an amazing tool with NEPER! 
I was wondering how it was possible to visualise crystal orientations with small cubes as shown with the example on neper.info.
The currently linked script.sh seems to relate to a different example and I could not find how to do this in the manual.
Thank you and best regards!

**Comment by rquey** [comment, marked as answer]
Here is an example:
```
neper -T -n 10 -morpho gg -statcell x,y,z,radeq -for tess,ori -orides rotmat
awk '{print $1,$2,$3}' n10-id1.stcell > cellcenters
awk '{print $4}' n10-id1.stcell > cellrads

paste cellrads n10-id1.ori | awk '{print 0.4*$1,$2,$3,$4,$5,$6,$7,$8,$9,$10}' > cubes

neper -V n10-id1.tess,cellcenters               \
          -showpoint 0                          \
          -dataedgerad 0.0012                   \
          -print tess                           \
          -showpoint 1                          \
          -showcell none                        \
          -showedge "domtype==1"                \
          -datapointrad "cube:file(cubes)"      \
          -datapointcol id                      \
          -datapointtrs 0.6                     \
          -datapointedgerad 0.00075             \
          -datapointedgecol "16|16|16"          \
          -print cubes
```
![image](https://user-images.githubusercontent.com/2659376/140778359-8ea68db3-722c-4141-a00f-dfe0d0326c82.png)
![image](https://user-images.githubusercontent.com/2659376/140778383-7ce415f9-c3d4-4fa7-9fb2-9cf99294d3e0.png)

**Reply by FelixTheTBone** [reply]
Many thanks for you help, I think I understand what code does and got it working!

**Comment by venki9494** [comment]
Please help me with similar code for Hexagonal orientation
